### PR TITLE
EDL & ExportIndicators enhancements

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21955
fixes: https://github.com/demisto/etc/issues/21957

## Description
Both integrations now support basic authentication.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 
- [ ] Code Review

## Technical Writer Review
- [ ] [EDL CHANGELOG](https://github.com/demisto/content/blob/80a7486105405d381691f9f70fd40725eda4260c/Integrations/EDL/CHANGELOG.md)
- [ ] [EDL YML](https://github.com/demisto/content/blob/80a7486105405d381691f9f70fd40725eda4260c/Integrations/EDL/EDL.yml) - line 58 needs to be reviewed

- [ ] [ExportIndicators CHANGELOG](https://github.com/demisto/content/blob/80a7486105405d381691f9f70fd40725eda4260c/Packs/ExportIndicators/Integrations/ExportIndicators/CHANGELOG.md)
- [ ] [ExportIndicators YML](https://github.com/demisto/content/blob/80a7486105405d381691f9f70fd40725eda4260c/Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml) - line 57 needs to be reviewed
- [ ] [ExportIndicators DOCS](https://github.com/demisto/content/blob/e083f2398dbc29f74b6f82734606f08b3e6f5fcb/Packs/ExportIndicators/Integrations/ExportIndicators/README.md) - lines 30-31 needs to be reviewed

### CODE REVIEW

**Comment by guyfreund** [comment]
@DeanArbel , @glicht FYI.

File: Packs/ExportIndicators/Integrations/ExportIndicators/README.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ This integration provides an endpoint with a list of indicators as a service for
     * __Private Key (Required for HTTPS)__: HTTPS private key provided by pasting its valuies into this field.
     * __HTTP Server__: Ignores certificate and private key, and will run the export indicators service
     in HTTP
+    * __Username__: The username to authenticate when fetching the indicators.
+    * __Password__: The username to authenticate when fetching the indicators.
```
Comment (Itay4) [inline Packs/ExportIndicators/Integrations/ExportIndicators/README.md]: fix description

File: Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -151,6 +167,16 @@ def route_edl_values() -> Response:
     Main handler for values saved in the integration context
     """
     params = demisto.params()
+
+    username: str = params.get('credentials', {}).get('identifier')
+    password: str = params.get('credentials', {}).get('password')
+    if username and password:
+        headers: dict = cast(Dict[Any, Any], request.headers)
```
Comment (Itay4) [inline Integrations/EDL/EDL.py]: i assume you did the cast cause of mypy?
what if we do some thing like `headers: Dict[Any, Any] =  request.headers` ?

File: Integrations/EDL/EDL.py [hunk 3]
```diff
@@ -151,6 +167,16 @@ def route_edl_values() -> Response:
...
+        if not validate_basic_authentication(headers, username, password):
+            err_msg: str = 'Basic authentication failed. Please make sure you are using the right credentials.'
```
Comment (Itay4) [inline Integrations/EDL/EDL.py]: no please, please

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml [hunk 4]
```diff
@@ -57,15 +49,17 @@ configuration:
...
-  display: HTTP Server
-  hidden: false
+- display: HTTP Server
   name: http_flag
   required: false
   type: 8
-description: Use the Export Indicators Service integration to provide an endpoint with a list of indicators as a service for
-  the system indicators.
+- display: Username
+  additionalinfo: Set the username which the user should enter when entering the indicators page.
```
Comment (Itay4) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: not sure if its a good description, lets work with @kirbles19  on that

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py [hunk 5]
```diff
@@ -170,6 +171,21 @@ def try_parse_integer(int_to_parse: Any, err_msg: str) -> int:
     return res
 
 
+def validate_basic_authentication(headers: dict, username: str, password: str) -> bool:
```
Comment (Itay4) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py]: add docstring

**Comment by glicht** [comment]
Can you remove the http checkbox in the configuration:

image

It should simply be if no certificate is provided then do http. No need for checkbox.

File: Integrations/EDL/EDL.yml [hunk 6]
```diff
@@ -54,6 +54,11 @@ configuration:
   name: http_flag
   required: false
   type: 8
+- display: Username
+  additionalinfo: Set the username which the user should enter when entering the indicators page.
```
Comment (glicht) [inline Integrations/EDL/EDL.yml]:
```suggestion
  additionalinfo: Enable basic authentication for accessing the EDL. If left empty, no authentication is enforced. 
```

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml (hunk 4 above)
Comment (glicht) [reply Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: See my suggestion on EDL

**Comment by Itay4** [comment]
lets change the header here, its way too big

image

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test.py [hunk 7]
```diff
@@ -244,3 +244,17 @@ def test_create_values_out_dict_4(self):
                 iocs_txt_json = json.load(iocs_txt_f)
                 for line in text_out.split('\n'):
                     assert line in iocs_txt_json
+
+    @pytest.mark.validate_basic_authentication
+    def test_validate_basic_authentication(self):
+        """Test Authentication"""
+        from ExportIndicators import validate_basic_authentication
+        username, password = 'user', 'pwd'
+        with open('ExportIndicators_test/TestHelperFunctions/authentication_test_data.json', 'r') as f:
```
Comment (glicht) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators_test.py]: I am not sure why you are doing this. Having the test data in a file is not necessary and actually just makes it harder to understand what is being tested. You can just pass the header that you want to test in the code.

**Comment by glicht** [comment]
Check if we can easily add the user agent to the logging of requests ( `User-agent` header).

When providing an invalid certificate (just enter garbage in private key) no meaningful error in the test module. The integration starts up in an invalid state.

File: Integrations/EDL/EDL.py [hunk 8]
```diff
@@ -224,8 +271,14 @@ def run_long_running(params):
         else:
             demisto.debug('Starting HTTP Server')
 
-        server = WSGIServer(('', port), APP, **ssl_args)
-        server.serve_forever()
+        server = WSGIServer(('', port), APP, **ssl_args, log=DEMISTO_LOGGER)
+        if is_test:
+            server_process = Process(target=server.serve_forever)
+            server_process.start()
+            time.sleep(10)
```
Comment (glicht) [inline Integrations/EDL/EDL.py]: Let's shorten this. Waiting 10 seconds is too much. Take this down to 5 seconds (should be sufficient).

**Comment by Itay4** [comment]
We might want to test the 5s timeout with an engine

**Comment by guyfreund** [comment]
Since checkout is in 20 min and we don't have a solution atm for https://github.com/demisto/etc/issues/22035, we'll merge the 2 issues mentioned up here.

**Comment by kirbles19** [comment]
@guyfreund done.

**Pull request merged**